src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -415,34 +415,50 @@ protected override int MakeSlot(BoundExpression node)
-                                //   struct A { B? B; }
-                                //   struct B { }
-                                //   if (a?.B != null) _ = ((A)a).B.Value; // no warning
-                                int containingSlot = MakeSlot(operand);
-                                return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operandType, containingSlot);
-                            }
+                            case ConversionKind.ExplicitNullable:
+                                {
+                                    var operand = conv.Operand;
+                                    var operandType = operand.Type;
+                                    var convertedType = conv.Type;
+                                    if (operandType?.IsNullableType() == true &&
+                                        convertedType?.IsNullableType() == false &&
+                                        operandType.GetNullableUnderlyingType().Equals(convertedType, TypeCompareKind.AllIgnoreOptions))
+                                    {
+                                        // Explicit conversion of Nullable<T> to T is equivalent to Nullable<T>.Value.
+                                        // For instance, in the following, when evaluating `((A)a).B` we need to recognize
+                                        // the nullability of `(A)a` (not nullable) and the slot (the slot for `a.Value`).
+                                        //   struct A { B? B; }
+                                        //   struct B { }
+                                        //   if (a?.B != null) _ = ((A)a).B.Value; // no warning
+                                        int containingSlot = MakeSlot(operand);
+                                        return containingSlot < 0 ? -1 : GetNullableOfTValueSlot(operandType, containingSlot);
+                                    }
+                                }
+                                break;
+                            case ConversionKind.Identity:
+                            case ConversionKind.ImplicitTupleLiteral:
+                            case ConversionKind.ImplicitTuple:
+                                if (isSupportedConversion(conv.Conversion, conv.Operand))
+                                {
+                                    return MakeSlot(conv.Operand);
+                                }
+                                break;
                         }
                     }
                     break;
                 case BoundKind.DefaultExpression:
                 case BoundKind.ObjectCreationExpression:
                 case BoundKind.DynamicObjectCreationExpression:
                 case BoundKind.AnonymousObjectCreationExpression:
-                    if (_placeholderLocalsOpt != null && _placeholderLocalsOpt.TryGetValue(node, out ObjectCreationPlaceholderLocal placeholder))
+                case BoundKind.TupleLiteral:
```

Comment:
>case BoundKind.TupleLiteral: [](start = 16, length = 28)

Can we reach this case (a `BoundTupleLiteral` that didn't get converted to a `BoundConvertedTupleLiteral`)?